### PR TITLE
Test: Replace t.error/fatal with assert/request in [raft_paper_test.go]

### PR DESCRIPTION
**Pull request by MrDXY**
Refactor: Replacing t.Error/t.Fatal with assert/require

This commit tries to eliminate boilerplate code in the testing suite by replacing occurrences
of t.Error and t.Fatal with assertions from the assert and require packages.

Changes Made:

- Replaced instances of t.Error with assert
- Replaced instances of t.Fatal with require

Affected Files:
- [raft_paper_test.go]

Part of: #146

### CODE REVIEW

File: raft_paper_test.go [hunk 1]
```diff
@@ -126,9 +120,7 @@ func TestLeaderBcastBeat(t *testing.T) {
 		{From: 1, To: 2, Term: 1, Type: pb.MsgHeartbeat},
 		{From: 1, To: 3, Term: 1, Type: pb.MsgHeartbeat},
 	}
-	if !reflect.DeepEqual(msgs, wmsgs) {
-		t.Errorf("msgs = %v, want %v", msgs, wmsgs)
-	}
+	assert.Equal(t, wmsgs, msgs)
```
Comment (pav-kv) [inline raft_paper_test.go:123]: could inline `wmsgs` as well:

```golang
assert.Equal(t, []pb.Message{
	{From: 1, To: 2, Term: 1, Type: pb.MsgHeartbeat},
	{From: 1, To: 3, Term: 1, Type: pb.MsgHeartbeat},
}, msgs)
```

Same in a few other places.

File: raft_paper_test.go [hunk 2]
```diff
@@ -164,24 +156,17 @@ func testNonleaderStartElection(t *testing.T, state StateType) {
...
-	}
-	if !r.trk.Votes[r.id] {
-		t.Errorf("vote for self = false, want true")
-	}
+	assert.Equal(t, uint64(2), r.Term)
+	assert.Equal(t, StateCandidate, r.state)
+	assert.True(t, r.trk.Votes[r.id])
+
 	msgs := r.readMessages()
 	sort.Sort(messageSlice(msgs))
```
Comment (pav-kv) [inline raft_paper_test.go:164]: For another PR: consider removing `messageSlice`, and using `slices.SortFunc` with some stable comparator.

**Review by pav-kv — APPROVED** [review]
LGTM % some suggestions

File: raft_paper_test.go (hunk 2 above)
Comment (MrDXY) [reply raft_paper_test.go:163]: Would this be what you had in mind?
```
slices.SortStableFunc(data, orderMessageAsc)
func orderMessageAsc(a pb.Message, b pb.Message) int {
	strA := fmt.Sprint(a)
	strB := fmt.Sprint(b)
	if strA > strB {
		return 1
	} else if strA < strB {
		return -1
	} else {
		return 0
	}
}
```

**Review by ahrtr — APPROVED** [review]
lgtm

Thanks

**Pull request merged**